app/models/concerns/searchable.rb

```diff
@@ -0,0 +1,72 @@
+require 'elasticsearch/model'
+
+module Searchable
+  extend ActiveSupport::Concern
+
+  included do
+    include Elasticsearch::Model
+
+    after_commit on: [:create] do
+      if Rails.env.test?
+        #reindex # call manually in tests if needed, for performance reasons
+      else
+        # TODO put on async queue for indexing
+      end
+    end
+
+    after_commit on: [:update] do
+      if Rails.env.test?
+        #reindex # call manually in tests if needed, for performance reasons
+      else
+        # TODO put on async queue for indexing
+      end
+    end
+
+    after_commit on: [:destroy] do
+      if Rails.env.test?
+        #remove_from_index # call manually in tests if needed, for performance reasons
+      else
+        # TODO put on async queue to remove_from_index
+      end
+    end
+
+    def reindex
+      __elasticsearch__.index_document
+      refresh_index
+      self
+    end
+
+    def remove_from_index
+      __elasticsearch__.destroy_document
+      refresh_index
+    end
+
+    def refresh_index
+      self.class.__elasticsearch__.refresh_index!
+    end
+
+    def self.rebuild_index
```

Comment:
What is this used for?  (Not called anywhere in this PR).  Is this the high-availability async reindexing?